types/index.d.ts

```diff
@@ -1000,6 +1001,11 @@ export declare namespace Knex {
     ): QueryBuilder<TRecord, TResult2>;
   }
 
+  interface OptimizerHint<TRecord extends {} = any, TResult extends {} = any> {
+    (...hints: string[]): QueryBuilder<TRecord, TResult>;
```

Comment:
I would prefer not to support this at all and just force user to pass array always

Reply:
@kibertoad You mean like either a single hint or an array of hints? Or you mean that even for a single hint an array with this single hint should be passed?

Reply:
Either would work. Either support a singular argument and wrap it into array inside, or always require an array; main thing I don't want to support is an arbitrary amount of separate arguments being passed and spread later.

Reply:
@kibertoad Changed in 65f488b